app/code/Magento/Email/Model/Transport.php

```diff
@@ -79,7 +116,17 @@ public function __construct(
             ScopeInterface::SCOPE_STORE
         );
 
-        $this->laminasTransport = new Sendmail($parameters);
+        $transport = $scopeConfig->getValue(
+            self::XML_PATH_TRANSPORT,
+            ScopeInterface::SCOPE_STORE
+        );
+
+        if ($transport === 'smtp') {
+            $this->laminasTransport = $this->createSmtpTransport($scopeConfig);
+        } else {
+            $this->laminasTransport = $this->createSendmailTransport($parameters);
```

Comment:
When the transport isn't SMTP, the behavior doesnt change.

Reply:
I don't think this behavior should be occurring in the constructor. I understand why it does (to change less code), but I think having the transport creation be interceptable would be beneficial

Reply:
Agreed, I tried to keep close to the original. So a `public function getTransport()` which does just this, and use that in the email?